File: agent/verticals/insurance.py
"""Insurance vertical definition."""

from agent.verticals.base import VerticalDefinition, tabs

VERTICAL = VerticalDefinition(
    key="insurance",
    label="Insurance",
    risk_level="high",
    entity_label_singular="plan",
    entity_label_plural="plans",
    default_plan_label="Insurance plan",
    crm_tabs=tabs(
        ("overview", "Overview"),
        ("readiness", "Readiness"),
        ("catalog", "Plans"),
        ("quote_flows", "Quotes"),
        ("leads", "Leads"),
        ("compliance", "Compliance"),
        ("activity", "Conversations"),
        ("prompt", "Prompt"),
        ("controls", "Controls"),
    ),
    entity_types=("insurance_plan", "insurer", "coverage_feature", "claim_flow", "document_requirement"),
    readiness_checks=("plans", "quote_flow", "claims", "renewals", "disclosures", "lead_capture"),
    action_types=(
        "SHOW_ENTITIES",
        "COMPARE_ENTITIES",
        "SORT_ENTITIES",
        "START_QUOTE",
        "REQUEST_CALLBACK",
        "CAPTURE_LEAD",
        "HANDOFF_TO_AGENT",
        "HANDOFF_TO_LICENSED_AGENT",
    ),
)

File: crm/src/verticals/definitions/insurance.ts
import { tab } from '../shared';
import type { CrmVerticalDefinition } from '../types';

export const insuranceVertical: CrmVerticalDefinition = {
  key: 'insurance',
  label: 'Insurance',
  riskLevel: 'high',
  entityLabelSingular: 'plan',
  entityLabelPlural: 'plans',
  defaultPlanLabel: 'Insurance plan',
  clientTabs: [
    tab('overview', 'Overview'),
    tab('readiness', 'Readiness'),
    tab('catalog', 'Plans'),
    tab('quote_flows', 'Quotes'),
    tab('leads', 'Leads'),
    tab('compliance', 'Compliance'),
    tab('activity', 'Conversations'),
    tab('prompt', 'Prompt'),
    tab('controls', 'Controls'),
  ],
  entityTypes: ['insurance_plan', 'insurer', 'coverage_feature', 'claim_flow', 'document_requirement'],
  readinessChecks: ['plans', 'quote_flow', 'claims', 'renewals', 'disclosures', 'lead_capture'],
  actionTypes: ['SHOW_ENTITIES', 'COMPARE_ENTITIES', 'SORT_ENTITIES', 'START_QUOTE', 'CAPTURE_LEAD', 'HANDOFF_TO_AGENT'],
};
